main.qml

```diff
@@ -1249,16 +1257,16 @@ ApplicationWindow {
             anchors.top: mobileHeader.bottom
             anchors.left: parent.left
             anchors.bottom: parent.bottom
-            onDashboardClicked: {middlePanel.state = "Dashboard"; if(isMobile) hideMenu()}
-            onTransferClicked: {middlePanel.state = "Transfer"; if(isMobile) hideMenu()}
-            onReceiveClicked: {middlePanel.state = "Receive"; if(isMobile) hideMenu()}
-            onTxkeyClicked: {middlePanel.state = "TxKey"; if(isMobile) hideMenu()}
-            onHistoryClicked: {middlePanel.state = "History"; if(isMobile) hideMenu()}
-            onAddressBookClicked: {middlePanel.state = "AddressBook"; if(isMobile) hideMenu()}
-            onMiningClicked: {middlePanel.state = "Mining"; if(isMobile) hideMenu()}
-            onSignClicked: {middlePanel.state = "Sign"; if(isMobile) hideMenu()}
-            onSettingsClicked: {middlePanel.state = "Settings"; if(isMobile) hideMenu()}
-            onKeysClicked: {settingsPasswordDialog.open(); if(isMobile) hideMenu()}
+            onDashboardClicked: { middlePanel.state = "Dashboard"; if(isMobile) hideMenu(); isBalanceVisible = false; updateBalance(); }
+            onTransferClicked: { middlePanel.state = "Transfer"; if(isMobile) hideMenu(); isBalanceVisible = false; updateBalance(); }
+            onReceiveClicked: { middlePanel.state = "Receive"; if(isMobile) hideMenu(); isBalanceVisible = true; updateBalance(); }
```

Comment:
The variable name/logic looks reversed here. 

Easy fix: rename var to `IsBalanceHidden`. 

On the other hand, after the refactoring this var becomes obsolete as you could do the check in `updateBalance()` instead. 

`middlePanel.balanceText = (middlePanel.state == "Receive") ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);`

Do as you prefer. =)